contracts/ngnx.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+//  ==========  External imports    ==========
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
+import "./helpers/ERC2771ContextUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+contract NGNX is
+    Initializable,
+    AccessControlUpgradeable,
+    ERC20Upgradeable,
+    ERC20PermitUpgradeable,
+    ERC2771ContextUpgradeable
+{
+    // Create a new role identifier for the minter role
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+
+    // --- ERC20 Data ---
+    string public constant version = "1";
+    uint public live; // Active Flag
+
+    // -- EVENTS --
+    event Cage();
+
+    function initialize(address[] memory trustedForwarder) public initializer {
+        __AccessControl_init();
+
+        __ERC20_init("NGNX Stablecoin", "NGNX");
+        __ERC20Permit_init("NGNX Stablecoin");
+
+        __ERC2771Context_init_unchained(trustedForwarder);
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        live = 1;
+    }
+
+    /**
+     * @dev sets a minter role
+     * @param account address for the minter role
+     */
+    function setMinterRole(address account) external {
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender));
+        _grantRole(MINTER_ROLE, account);
+    }
+
+    /**
+     * @dev Mints a new token
+     * @param account address to send the minted tokens to
+     * @param amount amount of tokens to mint
+     */
+    function mint(address account, uint amount) external returns (bool) {
+        require(live == 1, "NGNX/not-live");
+        require(hasRole(MINTER_ROLE, account));
+        _mint(account, amount);
+        return true;
+    }
+
+    /**
+     * @dev Burns a  token
+     * @param account address to burn tokens from
+     * @param amount amount of tokens to burn
+     */
+    function burn(address account, uint amount) external returns (bool) {
+        require(live == 1, "NGNX/not-live");
```

Comment:
Curious to know why you're checking for `require(live == 1, "NGNX/not-live");` when you already set live to be 1 in the initializer?

Reply:
The question is, wouldn't live be always equal to 1?

Reply:
live can be 0. when we want to stop the system

Reply:
There's no function to update the value of live. As it stands, live will always be 1 when deployed, unless the contract is upgraded later